File: types/Monster.ts
// types/Monster.ts

import { LeaderSkill } from "./LeaderSkill";

export interface Monster {
    id: number;
    name: string;
    stars: number;
    image?: string | null;
    element?: string | null;
    archetype?: string | null;
    nb_skill_ups?: number | null;
    leader_skill?: LeaderSkill | null;
    hp?: number | null;
    attack?: number | null;
    defense?: number | null;
    speed?: number | null;
    crit_rate?: number | null;
    crit_dmg?: number | null;
    resistance?: number | null;
    accuracy?: number | null;
}

File: backend/database/classes/monster.py
"""Classe Monster"""
from sqlalchemy import (
    Column,
    Integer,
    String,
    ForeignKey
)
from sqlalchemy.orm import relationship, mapped_column
from database.classes.base import SWdleBase

# Classe Monster
class Monster(SWdleBase):
    __tablename__ = 'monsters'

    name = Column(String(255), nullable=False)

    stars = Column(Integer, nullable=False)

    image = Column(String(255))

    element = Column(String(50))

    archetype = Column(String(50))

    nb_skill_ups = Column(Integer)

    leader_skill_id = mapped_column(ForeignKey("leader_skills.id"))
    leader_skill = relationship("LeaderSkill", back_populates="monsters")

    hp = Column(Integer)

    attack = Column(Integer)

    defense = Column(Integer)

    speed = Column(Integer)

    crit_rate = Column(Integer)

    crit_dmg = Column(Integer)

    resistance = Column(Integer)

    accuracy = Column(Integer)

    skills = relationship("MonsterSkill", back_populates="monster")
